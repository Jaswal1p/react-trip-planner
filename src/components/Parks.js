import React from 'react';

function Parks() {
    
    return (
        <div>
             
            <>
            <h2 role='exp'>
              <a  data-testid='link' href='/'>
                <span role='img' aria-label='happy' >ðŸ™‚</span>This is my Experience
              </a>
            
            </h2>
            
            </>
        </div>    
    );
}


export default Parks;