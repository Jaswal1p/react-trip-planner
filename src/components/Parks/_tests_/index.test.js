import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Parks from '../../Parks.js';

afterEach(cleanup);

describe('Parks component', () => {
    //First Test
    it('renders', () => {
        render(<Parks />);
      });

    // snapshot test 
    
    it('matches snapshot', () => {
       const { asFragment } = render(<Parks />);
       // assert value comparision

       expect(asFragment()).toMatchSnapshot();
    });

})

describe('emoji is visible', () => {
  it('inserts emoji into the h2', () => {

    // arrange test
    const { getByLabelText } = render(<Parks />);
    // const { getByRole } = render(<Parks />);

    // Assert 
    expect(getByLabelText('happy')).toHaveTextContent('ðŸ™‚');
    // expect(getByRole('img')).toHaveTextContent('ðŸ™‚')

  })
});

describe('links are visible', () => { 
  it('inserts text into the links2', () => {
       // arrange test
         const { getByTestId } = render(<Parks />);

       // Assert
       expect(getByTestId('link')).toHaveTextContent('This is my Experience');
  })
})