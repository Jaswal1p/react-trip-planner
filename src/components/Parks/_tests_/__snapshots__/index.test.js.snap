// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Parks component matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h2
      role="exp"
    >
      <a
        data-testid="link"
        href="/"
      >
        <span
          aria-label="happy"
          role="img"
        >
          ðŸ™‚
        </span>
        This is my Experience
      </a>
    </h2>
  </div>
</DocumentFragment>
`;
